### ISSUE TITLE
pachctl fsck --fix just stops - doesn't fix anything

### ISSUE THREAD

**Issue by ollystephens** [pachyderm/pachyderm#5768]
**What happened?**:

Running `pachctl fsck` reports a large number of subvenance errors.
Running `pachctl fsck --fix` reports the errors but then just stops; doesn't appear to fix.

Last output from the command is the last of the discovered consistency errors:

```
Error: consistency error: the commit was not in its provenance's subvenance
commit 1cd1cac04c3948918f2b5465ebb1bdd6 in repo sam-clean-ingress-documents
provenance commit 4828edbe81f14eb39399b1e89c3d6cd4 in repo ingress-dev-1-spout

Error: consistency error: the commit was not in its provenance's subvenance
commit 48c4f8b1b3f6458298135d85a00e4557 in repo sam-clean-ingress-documents
provenance commit e295177143d1416ba554b8cfecd437f9 in repo ingress-dev-1-spout
```

then the command exits with status 0. Running the command again issues the same set of consistency errors, demonstrating that they have not been fixed.

Logs of `pachd` and `etcd` show no relevant errors. 

**What you expected to happen?**:

Some sort of output to say the errors have been fixed? Or cannot be fixed?
Rerunning the command should find no errors assuming they can be fixed.

**How to reproduce it (as minimally and precisely as possible)?**:

Don't know how to recreate the specific scenario as not sure how it got like that.

**Anything else we need to know?**:

**Environment?**:
EKS cluster on AWS, running `v1.18.9-eks-d1db3d`
Pachyderm CLI and server running `1.12.4`
FSCK run from within the cluster in a job using the `pachyderm/pachyderm:1.12.4` image from docker hub.

**Comment by mindthevirt** [comment]
Hey @ollystephens, thanks for raising this issue. We're currently looking into it and will be in touch shortly.

**Comment by brendoncarroll** [comment]
I think it should be possible to fix this.  It looks like right now we fix missing commits, but not incorrect provenance/subvenance.  I'm working on a fix.

**Comment by ollystephens** [comment]
I noticed the fix had made it into 1.12.5 so I gave it a try and it was successful - fixed all of the reported issues. Thanks for adding that functionality!
